DSI/globus_gridftp_server_iRODS.cpp

```diff
@@ -1430,8 +1473,57 @@ globus_l_gfs_iRODS_command(
             }
             break;
 
+        case GLOBUS_GFS_CMD_RNTO:
+            {
+                namespace fs = irods::experimental::filesystem;
+                globus_gfs_log_message(GLOBUS_GFS_LOG_INFO,"iRODS: GLOBUS_GFS_CMD_RNTO\n");
+
+                if (cmd_info->from_pathname == nullptr)
+                {
+                    result = GlobusGFSErrorGeneric("iRODS: did not receive the from path");
+                    break;
+                }
+
+                char * from_path = strdup(cmd_info->from_pathname);
+                if (from_path == nullptr)
+                {
+                    result = GlobusGFSErrorGeneric("iRODS: strdup failed");
+                    break;
+                }
+                iRODS_l_reduce_path(from_path);
+                globus_gfs_log_message(GLOBUS_GFS_LOG_INFO,"rename from [%s] to [%s]\n", from_path, collection);
+
+                // determine if from_path is a collection or data object
+                dataObjCopyInp_t dataObjRenameInp;
+                try
+                {
+                   const auto object_status = fs::client::status(*iRODS_handle->conn, from_path);
+                    if (fs::client::is_data_object(*iRODS_handle->conn, from_path)) {
```

Comment:
Pass `object_status` to `is_data_object` instead of the connection and path.

So, `fs::client::is_data_object(object_status)`.